From 2a4a778d6543070c3eb053b30079a72dc943518f Mon Sep 17 00:00:00 2001
From: Li Jun <B47624@freescale.com>
Date: Mon, 1 Sep 2014 15:44:15 +0800
Subject: [PATCH] MLK-9617-2 usb: gadget: set bcdOTG of OTG descriptor for
 gadget drivers

This patch sets bcdOTG field of OTG descriptor for below 3 gadget drivers:
- ether
- mass storage
- serial
OTG and EH supplement release number in binary-coded decimal(i.e. 2.0 is 0200H).

Acked-by: Peter Chen <peter.chen@freescale.com>
Signed-off-by: Li Jun <b47624@freescale.com>
(cherry picked from commit a426ace68acd9b770ce3a609f92029782b8ca1d1)
---
 drivers/usb/gadget/ether.c        | 1 +
 drivers/usb/gadget/mass_storage.c | 1 +
 drivers/usb/gadget/serial.c       | 1 +
 3 files changed, 3 insertions(+)

diff --git a/drivers/usb/gadget/ether.c b/drivers/usb/gadget/ether.c
index e0722e9..b77946d 100644
--- a/drivers/usb/gadget/ether.c
+++ b/drivers/usb/gadget/ether.c
@@ -179,6 +179,7 @@ static struct usb_otg_descriptor otg_descriptor = {
 	 * it would not be called "OTG" ...
 	 */
 	.bmAttributes =		USB_OTG_SRP | USB_OTG_HNP,
+	.bcdOTG =		cpu_to_le16(0x0200),
 };
 
 static const struct usb_descriptor_header *otg_desc[] = {
diff --git a/drivers/usb/gadget/mass_storage.c b/drivers/usb/gadget/mass_storage.c
index 1178aea..d08c86d 100644
--- a/drivers/usb/gadget/mass_storage.c
+++ b/drivers/usb/gadget/mass_storage.c
@@ -73,6 +73,7 @@ static struct usb_otg_descriptor otg_descriptor = {
 	 * it would not be called "OTG" ...
 	 */
 	.bmAttributes =		USB_OTG_SRP | USB_OTG_HNP,
+	.bcdOTG =		cpu_to_le16(0x0200),
 };
 
 static const struct usb_descriptor_header *otg_desc[] = {
diff --git a/drivers/usb/gadget/serial.c b/drivers/usb/gadget/serial.c
index 0c7b826..51db0ee 100644
--- a/drivers/usb/gadget/serial.c
+++ b/drivers/usb/gadget/serial.c
@@ -87,6 +87,7 @@ static struct usb_otg_descriptor otg_descriptor = {
 	 * it would not be called "OTG" ...
 	 */
 	.bmAttributes =		USB_OTG_SRP | USB_OTG_HNP,
+	.bcdOTG =		cpu_to_le16(0x0200),
 };
 
 static const struct usb_descriptor_header *otg_desc[] = {
-- 
1.8.0

